test_data/nbc.html

```diff
@@ -0,0 +1,5 @@
cript src="https://nodeassets.nbcnews.com/_next/static/chunks/7903-5e5114a00bb0c279.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/3040-f57891343efde308.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/7711-90deb995f41b83f0.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/2062-b2a3d4e60f9c096a.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/7175-a2970e59b6376f6d.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/2402-aef5ed9e10e6f053.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/3501-76bb44e04fecc41b.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/2617-99777a35dbd32be2.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/8564-5941ee66d45f3a5c.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/6758-a30be510176d0df3.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/3833-d8b7a27424bd9270.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/6354-f75e8bc6c41261e6.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/3797-fc2b0063f2d25f68.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/3944-e288a77714e9cdcb.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/8089-16638f9cb98b03b9.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/9394-4aa84e31130a320d.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/470-f06132aecfe677e0.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/chunks/pages/article-675b0eb43572fc13.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/75b8d754a4609454425342e73043af569463f8c2/_buildManifest.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/75b8d754a4609454425342e73043af569463f8c2/_ssgManifest.js" defer=""></script><script src="https://nodeassets.nbcnews.com/_next/static/75b8d754a4609454425342e73043af569463f8c2/_middlewareManifest.js" defer=""></script></head><body class="articlePage news savory"><div id="modal-root" class="z-5 relative"></div><div id="__next" data-reactroot=""><div class="globalContainerStyles_container__XzPpd bg-knockout-primary" id="content"><div class="header-and-footer--banner-ad ad-container topbannerAd isScrolledToTop"><div class="ad dn-print"><div data-mps="true" data-slot="topbanner" data-sizes="[[[1000,1],[[728,90],[970,66],[970,90],[970,250],[1900,400]]],[[758,1],[[728,90],[970,66],[970,90],[970,250],[1900,400]]],[[0,0],[[1900,400]]]]" data-refresh-interval="0" data-render-on-view="true" data-targeting="{}" data-active-tab="true"></div></div></div><style>
```

Comment:
This particular example looks like a big chunk of JSON, though the date is in HTML <meta> tags. So apart from the meta tags, the rest of the doc isn’t parseable as HTML at all. Any chance we could use an example whose content can be parsed successfully as well? (I don’t believe it should be in scope for Crux to parse JSON-blob SPAs).

Reply:
That sounds like a pretty good simplification of this CL, plus it can easily be added later. Good idea to drop this!

Once you submit this PR, I will look into some basic validation for testing whether parsed dates are ISO 8601 or not (and convert if necessary).